diffrax/_brownian/tree.py

```diff
@@ -161,67 +273,183 @@ def _cond_fun(_state):
+                _bhh = _split_interval(_cond, _bhh_stu, _bhh_st_tu)
+                _bkk = None
+            else:
+                _bhh = None
+                _bkk = None
+
+            return _State(
+                level=_level,
+                s=_s,
+                w_s_u_su=_w,
+                key=_key,
+                bhh_s_u_su=_bhh,
+                bkk_s_u_su=_bkk,
+            )
 
         final_state = lax.while_loop(_cond_fun, _body_fun, init_state)
-        # Quadratic interpolation.
-        # We have w_s, w_t, w_u available to us, and interpolate with the unique
-        # parabola passing through them.
-        # Why quadratic and not just "linear from w_s to w_t to w_u"? Because linear
-        # only gets the conditional mean correct at every point, but not the
-        # conditional variance. This means that the Virtual Brownian Tree will pass
-        # statistical tests comparing w(t)|(w(s),w(u)) against the true Brownian
-        # bridge. (Provided s, t, u are greater than the discretisation level `tol`.)
-        # (If you just do linear then you find that the variance is *ever so slightly*
-        # too small.)
+
         s = final_state.s
-        u = final_state.u
-        w_s = final_state.w_s
-        w_t = final_state.w_t
-        w_u = final_state.w_u
-        rescaled_τ = (τ - s) / (u - s)
-        # Fit polynomial as usual.
-        # The polynomial ax^2 + bx + c is found by solving
-        # [s^2 s 1][a]   [w_s]
-        # [t^2 t 1][b] = [w_t]
-        # [u^2 u 1][c]   [w_u]
-        #
-        # `A` is the inverse of the above matrix, with s=0, t=0.5, u=1.
-        A = jnp.array([[2, -4, 2], [-3, 4, -1], [1, 0, 0]])
-        coeffs = jnp.tensordot(A, jnp.stack([w_s, w_t, w_u]), axes=1)
-        return jnp.polyval(coeffs, rescaled_τ)
-
-
-VirtualBrownianTree.__init__.__doc__ = """
-**Arguments:**
-
-- `t0`: The start of the interval the Brownian motion is defined over.
-- `t1`: The start of the interval the Brownian motion is defined over.
-- `tol`: The discretisation that `[t0, t1]` is discretised to.
-- `shape`: Should be a PyTree of `jax.ShapeDtypeStruct`s, representing the shape, 
-    dtype, and PyTree structure of the output. For simplicity, `shape` can also just 
-    be a tuple of integers, describing the shape of a single JAX array. In that case
-    the dtype is chosen to be the default floating-point dtype.
-- `key`: A random key.
-
-!!! info
-
-    If using this as part of an SDE solver, and you know (or have an estimate of) the
-    step sizes made in the solver, then you can optimise the computational efficiency
-    of the Virtual Brownian Tree by setting `tol` to be just slightly smaller than the
-    step size of the solver.
-
-The Brownian motion is defined to equal 0 at `t0`.
-"""
+        su = 2.0**-final_state.level
+
+        sr = r - s
+        ru = su - sr  # make sure su = sr + ru regardless of cancellation error
```

Comment:
Would replace these with
`sr = jax.nn.relu(r - s)
        ru = jax.nn.relu(
            su - sr
        )  # make sure su = sr + ru regardless of cancellation error`